cmd/guard-service/main.go

```diff
@@ -45,12 +49,74 @@ type learner struct {
 	pileLearnTicker *utils.Ticker
 }
 
+// Common method used for parsing ns, sid, cmFlag from all requests
+func (l *learner) baseHandler(query url.Values) (ns string, sid string, cmFlag bool, record *serviceRecord) {
```

Comment:
Why not return an error also in this function and that way deal with it from the caller's side?